src/main/kotlin/org/rust/toml/CratesIoApi.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml
+
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import com.google.gson.annotations.SerializedName
+import com.intellij.codeInsight.completion.CompletionUtil
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.application.ex.ApplicationUtil
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.psi.PsiElement
+import com.intellij.util.io.HttpRequests
+import org.jetbrains.annotations.TestOnly
+import org.rust.ide.notifications.showBalloon
+import org.toml.lang.psi.TomlKey
+import java.io.IOException
+
+data class SearchResult(val crates: List<CrateDescription>)
+
+data class CrateInfoResult(val crate: CrateDescription)
+
+data class CrateDescription(
+    val name: String,
+    @SerializedName("max_version")
+    val maxVersion: String
+)
+
+val CrateDescription.dependencyLine: String
+    get() = "$name = \"$maxVersion\""
+
+fun searchCrate(key: TomlKey): Collection<CrateDescription> {
+    MOCK?.let { return it }
```

Comment:
I like the way how we implement other mocks, for example, for extract function refactoring
More precisely, I like the error if `MOCK` variable in tests is null and error message.
Such error prevents you from accidental errors in tests and message says exactly what you should do
It would be great use such pattern here as well